sdk/proto/provider.proto

```diff
@@ -120,11 +120,23 @@ message DiffResponse {
     DiffChanges changes = 4;      // if true, this diff represents an actual difference and thus requires an update.
     repeated string diffs = 5;    // a list of the properties that changed.
 
+    map<string, DiffKind> richDiff = 6; // a rich diff appropriate for display.
```

Comment:
Any opinions on whether or not to make this a `map<string, DiffDetails>` where `DiffDetails` is a message rather than an enum value? Doing so would make it easy to add old + new property values in the future if necessary.

Reply:
FWIW, I think that would be very worthwhile.

Reply:
agree with matt on this.

Reply:
Done.